File: Data Structures & Algorithms/target-sum/submission-0.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:

        def dfs(i, curSum):
            if i == len(nums):
                if curSum == target:
                    return 1
                else:
                    return 0
            return dfs(i + 1, curSum + nums[i]) + dfs(i + 1, curSum - nums[i])

        return dfs(0, 0)
        

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer, Integer> numIndexMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int second = target - nums[i];
            if (numIndexMap.containsKey(second)) {
                result[0] = numIndexMap.get(second);
                result[1] = i;
                return result;
            }
            numIndexMap.put(nums[i], i);
        }
        return result;
        
    }
}
